File: javascript/008/main.js
/*
 * ProjectEuler 008
 * JavaScript / main.js
 * copyright (c) 2015 Susisu
 *
 * Node.js v0.12.0
 * `node main.js`
 */

var fs = require("fs");

function main() {
    var buffer = fs.readFileSync("./digits.txt");
    var digits = buffer.toString().split("\n")[0].split("").map(function (digit) { return parseInt(digit); });
    var max = 0;
    var n = 13;
    for (var i = 0; i <= digits.length - n; i++) {
        var product = 1;
        for (var j = 0; j < n; j++) {
            product *= digits[i + j];
        }
        if (product > max) {
            max = product;
        }
    }
    console.log(max);
}

main();

File: cpp/008/main.cpp
/*
 * ProjectEuler 008
 * C++ / main.cpp
 * copyright (c) 2015 Susisu
 *
 * Clang / Apple LLVM version 6.0 (clang-600.0.56) (based on LLVM 3.5svn)
 * `clang++ -std=c++11 -stdlib=libc++ main.cpp -o bin/main`
 */

#include <iostream>
#include <fstream>
#include <string>

#define N 13

int digitToInt(char c) {
    int n = c - '0';
    if (n < 0 || 10 <= n) {
        return 0;
    }
    else {
        return n;
    }
}

int main() {
    std::ifstream fs("./digits.txt");
    std::string digits;

    fs >> digits;

    int len = digits.length();
    long long max = 0;
    for (int i = 0; i <= len - N; i++) {
        long long prod = 1;
        for (int j = 0; j < N; j++) {
            prod *= digitToInt(digits.at(i + j));
        }
        if (prod > max) {
            max = prod;
        }
    }

    std::cout << max << std::endl;
    return 0;
}
